homeassistant/components/kaiterra/sensor.py

```diff
@@ -0,0 +1,257 @@
+from logging import getLogger
+from datetime import timedelta
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_DEVICES,
+    CONF_DEVICE_ID,
+    CONF_SCAN_INTERVAL,
+    CONF_TYPE,
+    CONF_NAME,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.util import Throttle
+
+_LOGGER = getLogger(__name__)
+
+AQI_SCALE = {
+    "cn": [0, 50, 100, 150, 200, 300, 400, 500],
+    "in": [0, 50, 100, 200, 300, 400, 500],
+    "us": [0, 50, 100, 150, 200, 300, 500],
+}
+
+AQI_LEVEL = {
+    "cn": [
+        {"label": "Good", "icon": "mdi:emoticon-excited"},
+        {"label": "Satisfactory", "icon": "mdi:emoticon-cool"},
+        {"label": "Moderate", "icon": "mdi:emoticon-happy"},
+        {"label": "Unhealthy for sensitive groups", "icon": "mdi:emoticon-neutral"},
+        {"label": "Unhealthy", "icon": "mdi:emoticon-sad"},
+        {"label": "Very unhealthy", "icon": "mdi:emoticon-dead"},
+        {"label": "Hazardous", "icon": "mdi:biohazard"}
+    ],
+    "in": [
+        {"label": "Good", "icon": "mdi:emoticon-excited"},
+        {"label": "Satisfactory", "icon": "mdi:emoticon-happy"},
+        {"label": "Moderately polluted", "icon": "mdi:emoticon-neutral"},
+        {"label": "Poor", "icon": "mdi:emoticon-sad"},
+        {"label": "Very poor", "icon": "mdi:emoticon-dead"},
+        {"label": "Severe", "icon": "mdi:biohazard"}
+    ],
+    "us": [
+        {"label": "Good", "icon": "mdi:emoticon-excited"},
+        {"label": "Moderate", "icon": "mdi:emoticon-happy"},
+        {"label": "Unhealthy for sensitive groups", "icon": "mdi:emoticon-neutral"},
+        {"label": "Unhealthy", "icon": "mdi:emoticon-sad"},
+        {"label": "Very unhealthy", "icon": "mdi:emoticon-dead"},
+        {"label": "Hazardous", "icon": "mdi:biohazard"}
+    ],
+}
+
+SENSORS = {
+    "rpm25c": {"name": "PM2.5", "icon": "mdi:weather-windy"},
```

Comment:
Please instead implement an `air_quality` platform for the air quality sensors.

Reply:
Ok, but how to attach temperature and humidity readings? via separate sensor entity?

Reply:
Those should be sensor entities in a sensor platform. So make a component module with the common logic and config schema. Then load the platforms, air_quality and sensor, via discovery. Or if including a config flow, forward the config entry to the platforms.

Reply:
It's ok to start with one platform but the backbone structure should be in place in the first PR to support multiple platforms.

Reply:
Component with separate platforms implemented.